### ISSUE TITLE
Patch: ignore chmod errors for filesystems not supporting the operation

### ISSUE THREAD

**Issue by agriggio** [BestImageViewer/geeqie#506]
I have most of my pictures on an external HD formatted with exFAT (for better interoperability). The current version of geeqie gives some annoying error messages when trying to perform some file operations (move, copy, rename), even though the op succeeds in the end. This is because `chmod` fails for the given file system. The following patch fixes the glitch, hopefully without introducing any regression.

```diff
diff --git a/src/ui_fileops.c b/src/ui_fileops.c
--- a/src/ui_fileops.c
+++ b/src/ui_fileops.c
@@ -483,7 +483,12 @@
 			/* Ignores chown errors, while still doing chown
 			   (so root still can copy files preserving ownership) */
 			ret = TRUE;
-			if (chmod(tl, st.st_mode) < 0) ret = FALSE;
+			if (chmod(tl, st.st_mode) < 0) {
+                            struct stat st2;
+                            if (stat(tl, &st2) != 0 || st2.st_mode != st.st_mode) {
+                                ret = FALSE;
+                            }
+                        }
 			}
 
 		tb.actime = st.st_atime;
```

Thanks for your great viewer btw!

**Comment by caclark** [comment]
I don't have a problem using a USB exfat drive. Is this perhaps a rights problem? Is your external drive being auto-mounted as root?

**Comment by agriggio** [comment]
I think I was getting something like this: https://groups.google.com/forum/m/#!topic/exfat/vqPCwJnA_Fo
might depend on the driver (or kernel) version.
do you think the proposed workaround is not good?

**Comment by caclark** [comment]
Fixed in commit 535b244d8 at geeqie.org
